internal/controllers/machine/machine_controller.go

```diff
@@ -706,23 +677,37 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
 		// Override the grace period of pods to reduce the time needed to skip them.
 		drainer.GracePeriodSeconds = 1
 
-		log.V(5).Info("Node is unreachable, draining will ignore gracePeriod. PDBs are still honored.")
+		log.V(3).Info("Node is unreachable, draining will use 1s GracePeriodSeconds and will ignore all Pods that have a deletionTimestamp > 1s old. PDBs are still honored.")
 	}
 
-	if err := kubedrain.RunCordonOrUncordon(drainer, node, true); err != nil {
+	if err := drainer.CordonNode(ctx, node); err != nil {
 		// Machine will be re-reconciled after a cordon failure.
-		log.Error(err, "Cordon failed")
-		return ctrl.Result{}, errors.Wrapf(err, "unable to cordon node %v", node.Name)
+		return ctrl.Result{}, errors.Wrapf(err, "failed to cordon Node %s", node.Name)
+	}
+
+	podDeleteList, err := drainer.GetPodsForEviction(ctx, nodeName)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	podsToBeDrained := podDeleteList.Pods()
+
+	if len(podsToBeDrained) == 0 {
+		log.V(2).Info("Drain completed, no Pods to drain on the Node")
+		return ctrl.Result{}, nil
 	}
 
-	if err := kubedrain.RunNodeDrain(drainer, node.Name); err != nil {
-		// Machine will be re-reconciled after a drain failure.
-		log.Error(err, "Drain failed, retry in 20s")
-		return ctrl.Result{RequeueAfter: 20 * time.Second}, nil
+	log.V(2).Info("Draining Node")
+
+	evictionResult := drainer.EvictPods(ctx, podDeleteList)
+	if evictionResult.DrainCompleted() {
+		log.V(2).Info("Drain completed, remaining Pods on the Node have been evicted")
+		return ctrl.Result{}, nil
 	}
 
-	log.Info("Drain successful")
-	return ctrl.Result{}, nil
+	conditions.MarkFalse(machine, clusterv1.DrainingSucceededCondition, clusterv1.DrainingNotCompletedReason, clusterv1.ConditionSeverityInfo, evictionResult.ConditionMessage())
```

Comment:
During a drain operation, as it organically fails during the initial runs, the evictionResult.ConditionMessage() would organically change (even though you made it deterministic) resulting in triggering continuous reconciles that supersedes the requeAfter period, correct? If so are we concern about this resulting in an overwhelming the kas with eviction API requests, which would keep increasing as you increase the number of Nodes you scale down at a time?

Reply:
> as it organically fails during the initial runs,

I assume with "fail" you mean that on the first runs the Pods won't just all immediately terminate, right? (just wondering if there are other cases that I'm missing)

> resulting in triggering continuous reconciles that supersedes the requeAfter period, correct

That is correct.

>  If so are we concern about this resulting in an overwhelming the kas with eviction API requests

In general, successful eviction requests are not repeated (because after the first one the Pod will have a deletionTimestamp). Still of course leaves the question about the ones that are failing because of e.g. a PDB.

Reply:
I'm going to add a logic to requeue if we already drained in the last x seconds (I'll think a bit about what x should be :))

Reply:
Fixed, PTAL :)